docs/API.md

```diff
@@ -1933,6 +1934,40 @@ a temporary location defined by `$postBuildInstallFromCargoBuildLogOut`
 
 **Required nativeBuildInputs**: assumes `cargo` is available on the `$PATH`
 
+### `craneLib.remapSourcePathPrefixHook`
+
+Defines `configureRustcRemapPathPrefix()` which can be used to set up a source
+map using the [`--remap-path-prefix`] option. The output of the derivation gains
+a dependency on the source code stored within the Nix store, but in return
+debuggers / other tools are able to resolve source file references to the
+correct files.
+
+The option is set by appending to the `CARGO_BUILD_RUSTFLAGS` environment variable,
+meaning it will not override any existing rustc flags you have set. However, the
+option may not be applied if you have set any `rustflags` configuration options
+with higher priority (e.g. target / `cfg()`-specific); in this case you will
+have to setup source mapping yourself.
+
+`configureRustcRemapPathPrefix()` takes three positional arguments:
+1. a path to where the source will map *to*, i.e. the project sources.
+   * If not specified, the value of `$src` will be used, if defined to a
+     non-empty value
+   * Otherwise, an error will be raised
+2. a path to where the source will map *from*, i.e. the current working
+   directory
+   * If not specified, the current working directory will be used
+3. `neuter` if remappings to Nix store paths should be "neutered" as part of the
+   build process (the default behavior)
+   * This removes the hash part of the store path from the build fingerprint,
+     which can be helpful to bypass unnecessary rebuilds resulting from usage of
+     e.g. `buildDepsOnly` / `mkDummySrc`
+
```

Comment:
The docs neglect to mention that running `configureRustcRemapPathPrefix` will schedule `fixupNeuteredRustcRemapPathPrefix` to later run and fixup the sources. I think for clarity the docs should cover `fixupNeuteredRustcRemapPathPrefix` as well so its clear what happens and when it happens

Reply:
This is great, thank you for addressing! Basically I wanted to make sure we outline what things run in what phases, but we don't necessarily have to spell out every little detail